evalml/tests/component_tests/test_arima_regressor.py

```diff
@@ -34,348 +34,247 @@ def test_model_instance(ts_data):
-
-    mock_get_dates.return_value = (X.index, X)
-    mock_format_dates.return_value = (X, y, None)
-
-    fh_ = forecasting.ForecastingHorizon(
-        [i + 1 for i in range(len(y))], is_relative=True
+    X_train, X_test, y_train = get_X_y(
+        train_features_index_dt,
+        train_target_index_dt,
+        train_none,
+        datetime_feature,
+        no_features,
+        test_features_index_dt,
     )
 
-    a_clf = sktime_arima.AutoARIMA()
-    clf = a_clf.fit(X=X, y=y)
-    y_pred_sk = clf.predict(fh=fh_, X=X)
-
-    m_clf = ARIMARegressor(d=None)
-    m_clf.fit(X=X, y=y)
-    mock_format_dates.return_value = (X, y, fh_)
-    y_pred = m_clf.predict(X=X)
-
-    assert (y_pred_sk == y_pred).all()
-
-
-@patch(
-    "evalml.pipelines.components.estimators.regressors.arima_regressor.ARIMARegressor._format_dates"
-)
-@patch(
-    "evalml.pipelines.components.estimators.regressors.arima_regressor.ARIMARegressor._get_dates"
-)
-def test_fit_predict_ts_with_X_not_y_index(
-    mock_get_dates, mock_format_dates, ts_data_seasonal_train
-):
-    X, y = ts_data_seasonal_train
-    assert isinstance(X.index, pd.DatetimeIndex)
-    assert isinstance(y.index, pd.DatetimeIndex)
-
-    mock_get_dates.return_value = (X.index, X)
-    mock_format_dates.return_value = (X, y, None)
-
     fh_ = forecasting.ForecastingHorizon(
-        [i + 1 for i in range(len(y))], is_relative=True
+        [i + 1 for i in range(len(X_test))], is_relative=True
     )
 
     a_clf = sktime_arima.AutoARIMA()
-    clf = a_clf.fit(X=X, y=y)
-    y_pred_sk = clf.predict(fh=fh_, X=X)
-
-    y = y.reset_index(drop=True)
-    assert not isinstance(y.index, pd.DatetimeIndex)
+    clf = a_clf.fit(X=X_train, y=y_train)
+    y_pred_sk = clf.predict(fh=fh_, X=X_test)
 
     m_clf = ARIMARegressor(d=None)
-    clf_ = m_clf.fit(X=X, y=y)
-    mock_format_dates.return_value = (X, y, fh_)
-    y_pred = clf_.predict(X=X)
-
-    assert (y_pred_sk == y_pred).all()
-
+    m_clf.fit(X=X_train, y=y_train)
+    y_pred = m_clf.predict(X=X_test)
 
-@patch(
-    "evalml.pipelines.components.estimators.regressors.arima_regressor.ARIMARegressor._format_dates"
-)
-@patch(
-    "evalml.pipelines.components.estimators.regressors.arima_regressor.ARIMARegressor._get_dates"
+    assert (y_pred_sk.values == y_pred.values).all()
+    assert y_pred.index.equals(X_test.index)
+
+
+@pytest.mark.parametrize(
+    "train_none, train_features_index_dt, "
+    "train_target_index_dt, no_features, "
+    "datetime_feature, test_features_index_dt",
+    [
+        (False, False, False, False, False, False),
+        (False, True, False, False, False, True),
+        (False, False, True, False, True, False),
+        (False, False, True, True, False, False),
+        (False, True, True, True, False, False),
+        (False, True, True, False, True, False),
+    ],
 )
-def test_fit_predict_ts_with_y_not_X_index(
-    mock_get_dates, mock_format_dates, ts_data_seasonal_train
+def test_fit_predict_sk_failure(
```

Comment:
nit: This is basically testing that without `remove_datetime`, our ARIMA component would fail right? Can we rename the test to reflect that and add a comment to the `pytest.raises` to explain why/what is happening?

Reply:
Actually this is testing a few different use cases in which the original `sktime.AutoARIMA` fails but our implementation succeeds.
- When the X index is of datetime type and the y index is not (and vice versa)
- When X has no features but still has an index
- When X has a datetime feature

So I named it `test_fit_predict_sk_failure` to represent the difference in behaviour from `sktime` and `evalml`